opsdroid/connector/matrix/connector.py

```diff
@@ -104,107 +104,130 @@ def _process_rooms_dict(self, rooms):
     @property
     def filter_json(self):
         """Define JSON filter to apply to incoming events."""
-        return {
-            "event_format": "client",
-            "account_data": {"limit": 0, "types": []},
-            "presence": {"limit": 0, "types": []},
-            "room": {"account_data": {"types": []}, "ephemeral": {"types": []}},
-        }
+        return json.dumps(
+            {
+                "event_format": "client",
+                "account_data": {"limit": 0, "types": []},
+                "presence": {"limit": 0, "types": []},
+                "room": {
+                    "account_data": {"types": []},
+                    "ephemeral": {"types": []},
+                    "state": {"types": []},
+                },
+            }
+        )
 
-    async def make_filter(self, api):
+    async def make_filter(self, api, fjson):
         """Make a filter on the server for future syncs."""
-        fjson = self.filter_json
-        resp = await api.create_filter(user_id=self.mxid, filter_params=fjson)
-        return resp["filter_id"]
+        path = "/_matrix/client/r0/user/" + self.mxid + "/filter"
+        headers = {"Authorization": "Bearer " + api.token}
+        resp = await api.send(method="post", path=path, data=fjson, headers=headers)
+
+        resp_json = await resp.json()
+        return resp_json["filter_id"]
 
     async def connect(self):
         """Create connection object with chat library."""
-        session = aiohttp.ClientSession(trust_env=True)
-        mapi = AsyncHTTPAPI(self.homeserver, session)
+        config = nio.ClientConfig(encryption_enabled=False, pickle_key="")
+        mapi = nio.AsyncClient(self.homeserver, self.mxid, config=config)
 
-        self.session = session
         login_response = await mapi.login(
-            "m.login.password", user=self.mxid, password=self.password
+            password=self.password, device_name=self.device_name
         )
-        mapi.token = login_response["access_token"]
+        if isinstance(login_response, nio.LoginError):
+            print("Error while logging in: " + login_response.message)
+
+        mapi.token = login_response.access_token
         mapi.sync_token = None
 
         for roomname, room in self.rooms.items():
-            response = await mapi.join_room(room["alias"])
-            self.room_ids[roomname] = response["room_id"]
+            response = await mapi.join(room["alias"])
+            if isinstance(response, nio.JoinError):
+                print("Error while joining room: " + roomname)
```

Comment:
Will opsdroid stop running if an exception is raised here? Then I think we should just log because joining the other rooms may work well.

Reply:
Yes I think it will. My preference would be to raise the error - some rooms may work but without further info from the user I think we have to assume that the bot needs to join every room in the config. (Some way to specify higher or lower priority of rooms might be a good way to address this, but that's a different PR.)